### ISSUE TITLE
Several versions of arm64 ios dylib failing to analyze with Index out of bounds errors (same build x86_64 analysis is OK)

### ISSUE THREAD

**Issue by clearbluejar** [NationalSecurityAgency/ghidra#6143]
**Describe the bug**

When analyzing any **arm64** version of dylib across Mac and iOS builds, receive several out of bounds index errors included below.  The same build binary on x86_64 analysis completes successfully. 

<html>
<body>

Analysis Task: Demangler GNU - Index 0 out of bounds for length 0
Analysis Task: Function Start Search After Code - Index 0 out of bounds for length 0
Analysis Task: Function Start Search After Data - Index 0 out of bounds for length 0
Analysis Task: Create Address Tables - Index 0 out of bounds for length 0
Analysis Task: Stack - Index 0 out of bounds for length 0

</body>
</html>

**To Reproduce**

Try analyzing one of these:
- Samples bins: 
[broken-arm64-dylibs-20240122T215313Z-001.zip](https://github.com/NationalSecurityAgency/ghidra/files/14015838/broken-arm64-dylibs-20240122T215313Z-001.zip)

- includes 2 arm64 bins that break analysis and one working x86_64 build

**Expected behavior**
Analysis would complete without index errors. 

**Screenshots**

```error.log
Index 0 out of bounds for length 0
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:361)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at ghidra.program.model.lang.protorules.HomogeneousAggregate.filter(HomogeneousAggregate.java:74)
	at ghidra.program.model.lang.protorules.ModelRule.assignAddress(ModelRule.java:162)
	at ghidra.program.model.lang.ParamListStandard.assignAddress(ParamListStandard.java:134)
	at ghidra.program.model.lang.ParamListStandard.assignMap(ParamListStandard.java:181)
	at ghidra.program.model.lang.PrototypeModel.assignParameterStorage(PrototypeModel.java:319)
	at ghidra.program.model.lang.PrototypeModel.getStorageLocations(PrototypeModel.java:363)
	at ghidra.program.database.function.FunctionDB.updateParametersAndReturn(FunctionDB.java:830)
	at ghidra.program.database.function.FunctionDB.loadVariables(FunctionDB.java:723)
	at ghidra.program.database.function.FunctionDB.getReturn(FunctionDB.java:375)
	at ghidra.program.database.function.FunctionDB.getSignatureSource(FunctionDB.java:2458)
	at ghidra.app.plugin.core.analysis.AbstractDemanglerAnalyzer.skipSymbol(AbstractDemanglerAnalyzer.java:212)
	at ghidra.app.plugin.core.analysis.AbstractDemanglerAnalyzer.demangleSymbols(AbstractDemanglerAnalyzer.java:125)
	at ghidra.app.plugin.core.analysis.AbstractDemanglerAnalyzer.doAdded(AbstractDemanglerAnalyzer.java:98)
	at ghidra.app.plugin.core.analysis.AbstractDemanglerAnalyzer.added(AbstractDemanglerAnalyzer.java:69)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:186)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:686)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:786)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:665)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:630)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:58)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:103)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:334)
	at java.base/java.lang.Thread.run(Thread.java:840)

---------------------------------------------------
Build Date: 2023-Dec-22 0936 EST
Ghidra Version: 11.0 (official GitHub distro)
JVM Version: Homebrew 17.0.9
OS: Mac OS X 14.2 aarch64

```

The other call stacks are similar. 

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
Build Date: 2023-Dec-22 0936 EST
Ghidra Version: 11.0 (official GitHub distro)
JVM Version: Homebrew 17.0.9
OS: Mac OS X 14.2 aarch64

**Comment by ghidracadabra** [comment]
This has been fixed in master: #6047

**Comment by clearbluejar** [comment]
great to know! thank you!🙏
